### PR TITLE
Add package-level documentation

### PR DESCRIPTION
**Pull request by tomaka**

### CODE REVIEW

File: packages/connect/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,101 @@
+/**
+ * The substrate-connect package makes it possible to connect to Substrate-compatible blockchains.
+ *
+ * Connecting to a chain is done in two steps:
+ *
+ * - Call {getConnectorClient} or {createPolkadotJsScClient}, which gives you a so-called *client*.
+ * - Call `addChain` or `addWellKnownChain` on this client.
```
Comment (wirednkod) [inline packages/connect/src/index.ts:7]:
```suggestion
 * 1 - Call {getConnectorClient} or {createPolkadotJsScClient}, which gives you a so-called *client*.
 * 2 - Call `addChain` or `addWellKnownChain` on returned client.
```

File: packages/connect/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+ * Failing to do so will lead to an error at the initialization of the parachain.
+ *
+ * Furthermore, the parachain must be added to the same client object as the one the relay chain
+ * was added to.
+ *
+ * In other words, this will work:
+ *
+ * ```js
+ * const client = getConnectorClient();
+ * await client.addChain(relayChain);
```
Comment (wirednkod) [inline packages/connect/src/index.ts:43]: Should there be mentioned in all these examples the `JSON.stringify(....)` or its irrelevant for current example-show-case?

File: packages/connect/src/index.ts [hunk 3]
```diff
@@ -1,3 +1,101 @@
...
+ *
+ * This means that there is no problem in calling {getConnectorClient} or
+ * {createPolkadotJsScClient} from within a library for example.
+ *
+ * # Well-known chains
+ *
+ * This package contains a list of so-called {WellKnownChains}. This is a list of popular chains
+ * that users are likely to connect to. Instead of calling `addChain` with a chain specification,
+ * one can call `addWellKnownChain`, passing only the name of a well-known chain as parameter.
+ *
```
Comment (wirednkod) [inline packages/connect/src/index.ts:80]:
```suggestion
 * At the moment these are polkadot, kusama, rococo and westend.
 * See <https://github.com/paritytech/substrate-connect/blob/main/packages/connect/src/WellKnownChains.ts>.
```

File: packages/connect/src/index.ts (hunk 2 above)
Comment (tomaka) [reply packages/connect/src/index.ts:43]: The point of this example is to illustrate what the text says, not to demonstrate how to use the library

File: packages/connect/src/index.ts (hunk 3 above)
Comment (tomaka) [reply packages/connect/src/index.ts:80]: Normally the JS docs generator should link to `WellKnownChains`, there's no reason to put a link.

**Review by wirednkod — APPROVED** [review]

**Pull request merged**